Competition2017/src/competition/subsystems/drive/commands/RotateToHeading.java

```diff
@@ -0,0 +1,51 @@
+package competition.subsystems.drive.commands;
+
+import com.google.inject.Inject;
+import xbot.common.command.BaseCommand;
+import xbot.common.math.ContiguousHeading;
+import xbot.common.math.PIDManager;
+import xbot.common.properties.XPropertyManager;
+import competition.subsystems.drive.DriveSubsystem;
+
+public class RotateToHeading extends BaseCommand{
+
+    DriveSubsystem drive;
+    ContiguousHeading targetHeading;
+    ContiguousHeading currentHeading;
+    
+    public final double defaultPValue = 1/80d;
+    double headingError;
+    
+    private PIDManager headingDrivePid;
+    
+    @Inject
+    public RotateToHeading(DriveSubsystem driveSubsystem, XPropertyManager propMan) {
+
+        this.drive = driveSubsystem;
+        headingDrivePid = new PIDManager("Rotate to heading", propMan, defaultPValue, 0, 0);
+    }
+
+    public void setTargetHeading(double heading) {
+        
+        targetHeading = new ContiguousHeading(heading);
+    }
+    
+    public void reset(){
+        
+        headingDrivePid.reset();
+    }
+    
+    @Override
+    public void initialize() {
+        
+    }
+
+    @Override
+    public void execute() {
+        
+        double errorInDegrees = drive.imu.getYaw().difference(targetHeading);
+        double rotationalPower = headingDrivePid.calculate(0, errorInDegrees);
+
+        drive.tankDrivePowerMode(rotationalPower, -rotationalPower);
+    }
```

Comment:
this command should isFinished when it's stable at it's target heading, otherwise we can't use it in auto to chain things.